torch/linalg/__init__.py

```diff
@@ -923,13 +923,15 @@
 """)
 
 matrix_rank = _add_docstr(_linalg.linalg_matrix_rank, r"""
-matrix_rank(A, tol=None, hermitian=False, *, out=None) -> Tensor
+linalg.matrix_rank(A, *, atol=None, rtol=None, hermitian=False, out=None) -> Tensor
+linalg.matrix_rank(A, tol, hermitian=False, *, out=None) -> Tensor
```

Comment:
How is this rendered in the docs? How do we want to advertise this? Perhaps just leave the new one but secretly support the old one? cc @mruberry

Reply:
There is actually an error when trying to build the docs using the current version. Probably there should be an empty line.
This double function signature was inspired by [Julia docs for pinv](https://docs.julialang.org/en/v1/stdlib/LinearAlgebra/#LinearAlgebra.pinv).
I think we should leave the new signature as main one and in the description tell that there is also NumPy compatible version.

Reply:
Completely agree. I'd say we go with that.

Reply:
Also, about the Numpy compat, see https://github.com/pytorch/pytorch/pull/63102#discussion_r689158004